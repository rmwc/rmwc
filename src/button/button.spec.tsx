import React from 'react';
import { render, screen } from '@testing-library/react';
import { Button } from './';

describe('Button', () => {
  it('renders', () => {
    const { asFragment } = render(<Button>Button</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('can have an icon', () => {
    const { asFragment } = render(<Button icon="favorite">Button</Button>);
    expect(screen.getByText('favorite')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('can be raised', () => {
    const { container } = render(<Button raised />);
    expect(container.firstChild).toHaveClass('mdc-button--raised');
  });

  it('can be danger', () => {
    const { asFragment } = render(<Button danger />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('can be unelevated', () => {
    const { container } = render(<Button unelevated />);
    expect(container.firstChild).toHaveClass('mdc-button--unelevated');
  });

  it('can be outlined', () => {
    const { container } = render(<Button outlined />);
    expect(container.firstChild).toHaveClass('mdc-button--outlined');
  });

  it('can be dense', () => {
    const { container } = render(<Button dense />);
    expect(container.firstChild).toHaveClass('mdc-button--dense');
  });

  it('can ripple', () => {
    const { container } = render(<Button ripple />);
    expect(container.firstChild).toHaveClass('mdc-button');
  });

  it('can not ripple', () => {
    const { container } = render(<Button ripple={undefined} />);
    expect(container.firstChild).toHaveClass('mdc-button');
  });

  it('can have custom classnames', () => {
    const { container } = render(<Button className={'my-custom-classname'} />);
    expect(container.firstChild).toHaveClass('my-custom-classname');
  });
});
