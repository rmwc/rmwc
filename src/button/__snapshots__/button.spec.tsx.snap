// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button can be danger 1`] = `
<DocumentFragment>
  <button
    class="mdc-button"
    style="--mdc-theme-primary: var(--mdc-theme-error); --mdc-theme-on-primary: var(--mdc-theme-on-error);"
  >
    <div
      class="mdc-button__ripple"
    />
    <span
      class="mdc-button__label"
    />
  </button>
</DocumentFragment>
`;

exports[`Button can have an icon 1`] = `
<DocumentFragment>
  <button
    class="mdc-button"
  >
    <div
      class="mdc-button__ripple"
    />
    <i
      class="rmwc-icon rmwc-icon--ligature material-icons mdc-button__icon"
    >
      favorite
    </i>
    <span
      class="mdc-button__label"
    >
      Button
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button renders 1`] = `
<DocumentFragment>
  <button
    class="mdc-button"
  >
    <div
      class="mdc-button__ripple"
    />
    <span
      class="mdc-button__label"
    >
      Button
    </span>
  </button>
</DocumentFragment>
`;
